### DISCUSSION TITLE
Getting 404 when loading dynamic routes on nextjs

### DISCUSSION THREAD

**Discussion by Tanjeeb** [vercel/next.js#23503] [Help] [answered]
I make a dynamic route like [results].js 
```
import { useRouter } from 'next/router'

const Results = () => {
    const router = useRouter();
    const { results } = router.query;

    return <p>Post: {results}</p>
}

export default Results
```
and configure my next.confi.js

```
module.exports = {
    trailingSlash: true,
}

```
It's working well on dev mode but in production mode, I have the only problem with the dynamic page route after npm run export.When I brows htttps://my-domain/search/searchslog 
it's getting 404 error page.

**Comment by akellbl4** [comment, marked as answer]
Hi,
Have you checked this [docs](https://nextjs.org/docs/advanced-features/static-html-export)?
The fact is after static export you couldn't use dynamic routes.

Actually, you could solve your problem with any kind of proxy server. You can route queries like `/search/searchslog` to file `/search/[results].js`  by matching pathname to the file. I'm doing a similar thing on one of my projects.

**Reply by Tanjeeb** [reply]
can you please tell me what will be the getStaticPaths and getStaticProps for dynamic search(/search/[results].js)?

**Reply by akellbl4** [reply]
if your `result` is an unpredictable value you aren't able to generate all of the results for the search. In this case, you should use `getServerSideProps` or if you'd like to go full static I can suggest matching your param with the query string.

For example:
```jsx
// `/pages/search.js`
import Router, { useRouter } from "next/router";

export default function Search() {
  const { query, asPath } = useRouter();
  const { results } = query;

  function handleSubmit(evt) {
    const results = new FormData(evt.currentTarget).get("search");
    const pathname = asPath.split("?")[0];
    const url = { pathname, query: { results } };

    evt.preventDefault();
    Router.push(url, url, { shallow: true });
  }

  console.log(query);

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <input type="text" name="search" />
        <button type="submit">search</button>
      </form>
      {results && <p>Post: {results}</p>}
    </div>
  );
}
```

In other words, use query string instead of dynamic routes for static export.
